libs/nss.c

```diff
@@ -100,12 +108,6 @@ int _nss_tcb_getspent_r(struct spwd *__result_buf,
 	off_t currpos;
 	int retval, saved_errno;
 
-	if (!tcbdir) {
```

Comment:
What should happen in this function if _nss_tcb_setspent failed to open the directory?

Reply:
From my understanding of [the documentation](https://www.gnu.org/software/libc/manual/html_node/NSS-Module-Function-Internals.html): In this case the function shouldn't even be called through the high-level NSS interface, as we already signaled during setspent, that information is not available.

Reply:
I think @ldv-alt has a good point. If we're trying to have defensive programming against the caller's mistakes, we should keep this piece in `_nss_tcb_getspent_r`:

```c
	if (!tcbdir)
		return NSS_STATUS_UNAVAIL;
```

Reply:
Addressed in rebased commit.